pkg/apis/core/validation/capabilitles.go

```diff
@@ -0,0 +1,352 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package validation
+
+import (
+	"fmt"
+	"strings"
+
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/apimachinery/pkg/util/json"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/utils/ptr"
+
+	"github.com/gardener/gardener/pkg/apis/core"
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+)
+
+// IsDefined checks if the capabilitiesDefinition is set and not empty
+// it is intended to be used only during the transition period to capabilities and should be removed after capabilitiesDefinition is required
+// then only validateCapabilitiesDefinition should be used
+func IsDefined(capabilitiesDefinition *core.Capabilities) bool {
+	return capabilitiesDefinition != nil && len(*capabilitiesDefinition) != 0
+}
```

Comment:
If the `Capabilities` are passed plainly, we can subsequently change this method as well:

```suggestion
func IsDefined(capabilitiesDefinition core.Capabilities) bool {
	return len(capabilitiesDefinition) != 0
}
```